engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -842,6 +878,707 @@ static int SetTexture(lua_State* L)
+    lua_rawset(L, -3);
+
+    lua_pushliteral(L, "animations");
+    lua_newtable(L);
+
+    uint32_t num_geometries = texture_set->m_Geometries.m_Count;
+
+    for (int i = 0; i < texture_set->m_Animations.m_Count; ++i)
+    {
+        dmGameSystemDDF::TextureSetAnimation& anim = texture_set->m_Animations[i];
+
+        // Note:
+        // frame_start and frame_end is not necessarily the same thing as an animations m_Start/m_End,
+        // since we generate the geometry based on input textures. But now with the
+        // resource.set_atlas function we allow creating arbitrary geometry and mapping
+        // that to animations and have no concept of input textures.
+        // So we need this indirection to be able to support both creating custom animations in runtime
+        // and the way we have created the DDF in the build pipeline.
+        uint32_t index_start = texture_set->m_FrameIndices[anim.m_Start];
+        uint32_t index_end   = index_start + (anim.m_End - anim.m_Start);
+
+        lua_pushinteger(L, (lua_Integer) (i+1));
+        lua_newtable(L);
+
+        lua_pushliteral(L, "id");
+        lua_pushstring(L, anim.m_Id);
+        lua_rawset(L, -3);
+
+        #define SET_LUA_VAL(set_fn, id, val) \
+            lua_pushliteral(L, id); \
+            set_fn(L, val); \
+            lua_rawset(L, -3);
+
+        SET_LUA_VAL(lua_pushinteger, "width", anim.m_Width);
+        SET_LUA_VAL(lua_pushinteger, "height", anim.m_Height);
+        SET_LUA_VAL(lua_pushinteger, "fps", anim.m_Fps);
+        SET_LUA_VAL(lua_pushinteger, "playback", (lua_Integer) DDFPlaybackToGameObjectPlayback(anim.m_Playback));
+        SET_LUA_VAL(lua_pushinteger, "frame_start", index_start + 1);
+        SET_LUA_VAL(lua_pushinteger, "frame_end", index_end + 1);
+        SET_LUA_VAL(lua_pushboolean, "flip_horizontal", anim.m_FlipHorizontal);
+        SET_LUA_VAL(lua_pushboolean, "flip_vertical", anim.m_FlipVertical);
+
+        #undef SET_LUA_VAL
+
+        lua_rawset(L, -3);
+    }
+
+    lua_rawset(L, -3);
+
+    {
+        lua_pushliteral(L, "geometries");
+        lua_newtable(L);
+
+        for (int i = 0; i < num_geometries; ++i)
+        {
+            dmGameSystemDDF::SpriteGeometry& geom = texture_set->m_Geometries[i];
+
+            lua_pushinteger(L, (lua_Integer) (i+1));
+            lua_newtable(L);
+
+            {
+                assert(geom.m_Vertices.m_Count % 2 == 0);
+                assert(geom.m_Uvs.m_Count      % 2 == 0);
+                assert(geom.m_Indices.m_Count  % 3 == 0);
+
+                lua_pushliteral(L, "vertices");
+                lua_newtable(L);
+                for (int j = 0; j < geom.m_Vertices.m_Count; j += 2)
+                {
+                    float x = (geom.m_Vertices[j] + 0.5) * geom.m_Width;
+                    float y = (0.5 - geom.m_Vertices[j+1]) * geom.m_Height;
+
+                    lua_pushinteger(L, j + 1);
+                    lua_pushnumber(L, x);
+                    lua_rawset(L, -3);
```

Comment:
I think you should be able to use [lua_rawseti](https://www.lua.org/manual/5.1/manual.html#lua_rawseti) to make the code slightly short (imho a little more readable)

SImilarly, for string keys, there's `lua_setfield()`